search-app/src/cmr/search/services/parameters/converters/shapefile.clj

```diff
@@ -114,6 +117,50 @@
         (errors/throw-service-error :bad-request (format "Cannot transform source CRS [%s] to WGS 84" src-crs-name))))
     geometry))
 
+(defn feature-point-count
+  "Get the number of points in the Feature"
+  [feature]
+  (let [crs (when (.getDefaultGeometryProperty feature)
+              (-> feature .getDefaultGeometryProperty .getDescriptor .getCoordinateReferenceSystem))
+        properties (.getProperties feature)
+        geometry-props (filter (fn [p] (geo/geometry? (.getValue p))) properties)
+        geometries (map #(-> % .getValue (transform-to-epsg-4326 crs)) geometry-props)]
+    (apply + (map geometry-point-count geometries))))
+
+(defn features-point-count
+  "Compute the number of points in a list of Features"
+  [features]
+  (reduce (fn [total feature] (+ total (feature-point-count feature)))
+          0
+          features))
+
+(defn validate-point-count
+  "Validate that the number of points in the features is greater than zero and less than the limit"
+  [features]
+  (let [point-count (features-point-count features)]
+    (if (= point-count 0)
```

Comment:
This might be more readable with a `cond`

Reply:
Agreed, done.